table-module/src/main/java/com/iluwatar/tablemodule/RoomDaoImpl.java

```
@@ -0,0 +1,212 @@
+   */
+  @Override
+  public Boolean add(final Room room) throws Exception {
+    if (getById(room.getId()).isPresent()) {
+      return false;
+    }
+
+    try (var connection = getConnection();
+         var statement = connection.prepareStatement("INSERT INTO ROOMS VALUES (?,?,?,?)")) {
+      statement.setInt(1, room.getId());
+      statement.setString(2, room.getRoomType());
+      statement.setInt(3, room.getPrice());
+      statement.setBoolean(4, room.isBooked());
+      statement.execute();
+      return true;
+    } catch (Exception e) {
+      throw new Exception(e.getMessage(), e);
+    }
+  }
+
+  /**
+   * Update the room information in the database.
+   *
+   * @param room the room to be updated
+   * @return the status of this operation
+   * @throws Exception if any error
+   */
+  @Override
+  public Boolean update(final Room room) throws Exception {
+    try (var connection = getConnection();
+         var statement =
+             connection
+                 .prepareStatement("UPDATE ROOMS SET ROOM_TYPE = ?, PRICE = ?, BOOKED = ?"
+                     + " WHERE ID = ?")) {
+      statement.setString(1, room.getRoomType());
+      statement.setInt(2, room.getPrice());
+      statement.setBoolean(3, room.isBooked());
+      statement.setInt(4, room.getId());
+      return statement.executeUpdate() > 0;
+    } catch (Exception e) {
+      throw new Exception(e.getMessage(), e);
+    }
+  }
+
+  /**
+   * Delete a room from the database.
+   *
+   * @param room the room to be deleted
+   * @return the status of this operation
+   * @throws Exception if any error
+   */
+  @Override
+  public Boolean delete(final Room room) throws Exception {
+    try (var connection = getConnection();
+         var statement = connection.prepareStatement("DELETE FROM ROOMS WHERE ID = ?")) {
+      statement.setInt(1, room.getId());
+      return statement.executeUpdate() > 0;
+    } catch (Exception e) {
+      throw new Exception(e.getMessage(), e);
+    }
```

Maybe just let it throw

---

Please check the same issue elsewhere too